kernel: mvebu: add Linux 5.15 support

rsalvaterra opened the pull request:
Tentatively removing the WIP tag. Split into logical steps.

Build-tested:
mvebu/cortexa53
mvebu/cortexa72

Run-tested:
mvebu/cortexa9 (Turris Omnia)

~~NOTE: The PCIe Aardvark controller driver is **broken**. Its patches are in generic and they were most likely badly rebased. For this reason, the cortexa53 subtarget is failing to build.~~

With @Ansuel's fix, the cortexa53 subtarget is now building.

rsalvaterra commented:
@Ansuel, any ideas about the Aarvark driver?

robimarko commented:
Those aardvark PCI patches in pending-5.15 can probably be dropped, most of the should already be in 5.15 and there is probably a good reason for those that did not make it even in 5.17

anomeome commented:
Building to mvebu:cortexa9 kernel config fault followed by mwlwifi [build error](https://termbin.com/9fsc)

anomeome commented:
Started from a _dirclean_, so do not think what I'm seeing is some kind of dried muffin remnant. A [diffconfig](https://termbin.com/nrpb) as possibly idiosyncratic config issue.

rsalvaterra commented:
> @rsalvaterra can you pass me the build error? Probably target/linux/refresh made some big funny thing :(

@Ansuel, here's the error: https://gist.github.com/rsalvaterra/d112a5df2c7418dc47c1866117635b27. It's complaining that `virq` hasn't been declared. I guess the declaration was probably dropped in some badly rebased patch…?

> Those aardvark PCI patches in pending-5.15 can probably be dropped, most of the should already be in 5.15 and there is probably a good reason for those that did not make it even in 5.17

@robimarko, if true, that would be great, but unfortunately I don't have the hardware to test that hypothesis.

Ansuel commented:
yes that seems to be a bad refreshed patch.. 99% some bad output of fixing patch fuzz should be an easy fix honestly but i think the correct one would be backporting the upstream version 
will check and report here

1Jeff1 commented:
Yeah https://github.com/openwrt/openwrt/pull/9592 has fixed the buid error with the cortexa53 subtarget, I testing on GL-MV1000,it seems that there is other bug, no internet connection, weird, as I can ping from within the router and everything looks normal. https://gist.github.com/1Jeff1/1de41d2d3290f3796a10ab462149f0b3

robimarko commented:
@anomeome This will get mwlwifi working under 5.15
https://github.com/openwrt/openwrt/pull/9595

It does however still have warnings when you disconnect clients or reboot the AP.

anomeome commented:
PR9595 resolves mwlwifi compilation issue. Built and flashed to a mamba with [boot failure](https://termbin.com/a5bs)

lucize commented:
maybe with someone can also fix ?

```
Package kmod-usb-net-asix is missing dependencies for the following libraries:
mdio_devres.ko
selftests.ko

```

njhsi commented:
dmesg says:
```
 25.654289] mv88e6085 f1072004.mdio-mii:00: port 3 failed to add 12:cb:04:50:e9:bd vid 7 to fdb: -95
[   25.738280] mv88e6085 f1072004.mdio-mii:00: port 3 failed to add 12:cb:04:50:e9:bd vid 5 to fdb: -95
```
might be good just to silence these errors.

Ansuel commented:
@njhsi these are errors from dsa... totally not something to ignore and probably the cause of the network problems :D

rsalvaterra committed:
mvebu: copy 5.10 patches to 5.15

Linux 5.15 on Marvell EBU, here we go!

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
mvebu: remove all ≤ 5.15-tagged patches

They're already upstream.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
mvebu: copy 5.10 kconfigs to 5.15

Will be refreshed/updated later.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
mvebu: update the kconfigs for 5.15

And remove irrelevant stuff while at it.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
mvebu: add Linux 5.15 as testing kernel

Run-tested on the cortexa9 subtarget (Turris Omnia).

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

Ansuel commented:
@robimarko remember the change for nfs_ssc?

robimarko commented:
Yeah, why?

Eliminater74 commented:
When I try, here is the log, of the error I get: 

make\[3\]: Entering directory '/workdir/openwrt/package/kernel/linux'  
mkdir -p /workdir/openwrt/staging\_dir/target-arm\_cortex-a9+vfpv3-d16\_musl\_eabi/root-mvebu/stamp

SHELL= flock /workdir/openwrt/tmp/.root-copy.flock -c 'cp -fpR /workdir/openwrt/build\_dir/target-arm\_cortex-a9+vfpv3-d16\_musl\_eabi/linux-mvebu\_cortexa9/packages/.pkgdir/kernel/. /workdir/openwrt/staging\_dir/target-arm\_cortex-a9+vfpv3-d16\_musl\_eabi/root-mvebu/'  
touch /workdir/openwrt/staging\_dir/target-arm\_cortex-a9+vfpv3-d16\_musl\_eabi/root-mvebu/stamp/.kernel\_installed  
ERROR: module '/workdir/openwrt/build\_dir/target-arm\_cortex-a9+vfpv3-d16\_musl\_eabi/linux-mvebu\_cortexa9/linux-5.15.31/fs/nfs\_common/nfs\_ssc.ko' is missing.  
make\[3\]: \*\*\* \[modules/fs.mk:420: /workdir/openwrt/bin/targets/mvebu/cortexa9/packages/kmod-fs-nfs-common\_5.15.31-1\_arm\_cortex-a9\_vfpv3-d16.ipk\] Error 1  
make\[3\]: Leaving directory '/workdir/openwrt/package/kernel/linux'  
time: package/kernel/linux/compile#2.84#0.19#4.35

For now, ill disable all NFS Kmods and see what happends.

Ansuel commented:
it was to fix this error but looking at the makefile it doesn't make sense

robimarko commented:
Yeah, but the change you made, made no sense.
`CONFIG_NFS_V4_2_SSC_HELPER` is `default y if NFS_V4_2` so it should be autoselected.

To rule this out, just add `CONFIG_NFS_V4_2_SSC_HELPER=y` to the `kmod-fs-nfs-common` KCONFIG

rsalvaterra commented:
Hm… guys? Is there any problem with this pull request building with the **default** configuration? Kernel modules and packages are a different issue.

anomeome commented:
Flashed and testing on a mamba (with PR9595) everything LGTM.

1Jeff1 commented:
@anomeome does the LAN ports work?, testing on wrt32x has the same DSA problems as on the mv-1000, over wifi seems to work ok.

anomeome commented:
Have not seen anything anomalous relative to 5.10 image built with same config / patches, and device config setup as currently utilised. I will note that I am deleting the mvebu 700-patch and replacing MQ with MQPrio at boot. Have not tried flashing the rango I have as I assumed from the above that others were testing. I will take a kick at that device when I can free it up.

Eliminater74 commented:
Error: seems to fall under "`mwlwifi"`

```
1297:10: error: implicit declaration of function 'get_fs'; did you mean 'sget_fc'? [-Werror=implicit-function-declaration]
 1297 |  oldfs = get_fs();
      |          ^~~~~~
      |          sget_fc
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1297:10: error: incompatible types when assigning to type 'mm_segment_t' from type 'int'
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1298:2: error: implicit declaration of function 'set_fs'; did you mean 'sget_fc'? [-Werror=implicit-function-declaration]
 1298 |  set_fs(KERNEL_DS);
      |  ^~~~~~
      |  sget_fc
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1298:9: error: 'KERNEL_DS' undeclared (first use in this function); did you mean 'KERNFS_NS'?
 1298 |  set_fs(KERNEL_DS);
      |         ^~~~~~~~~
      |         KERNFS_NS
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1298:9: note: each undeclared identifier is reported only once for each function it appears in
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c: At top level:
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:34:23: error: expected declaration specifiers or '...' before string constant
   34 | #define PCIE_DEV_NAME "Marvell 802.11ac PCIE Adapter"
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1644:25: note: in expansion of macro 'PCIE_DEV_NAME'
 1644 | MODULE_SUPPORTED_DEVICE(PCIE_DEV_NAME);
      |                         ^~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:277: /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.o] Error 1
make[4]: *** [Makefile:1868: /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb] Error 2
make[4]: Leaving directory '/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/linux-5.15.31'
make[3]: *** [Makefile:103: /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/.built] Error 2
make[3]: Leaving directory '/workdir/openwrt/package/kernel/mwlwifi'
time: package/kernel/mwlwifi/compile#1.35#0.24#2.71
```

For Device wrt3200acm

rsalvaterra commented:
> Error: seems to fall under "`mwlwifi"`
> 
> ```
> 1297:10: error: implicit declaration of function 'get_fs'; did you mean 'sget_fc'? [-Werror=implicit-function-declaration]
>  1297 |  oldfs = get_fs();
>       |          ^~~~~~
>       |          sget_fc
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1297:10: error: incompatible types when assigning to type 'mm_segment_t' from type 'int'
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1298:2: error: implicit declaration of function 'set_fs'; did you mean 'sget_fc'? [-Werror=implicit-function-declaration]
>  1298 |  set_fs(KERNEL_DS);
>       |  ^~~~~~
>       |  sget_fc
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1298:9: error: 'KERNEL_DS' undeclared (first use in this function); did you mean 'KERNFS_NS'?
>  1298 |  set_fs(KERNEL_DS);
>       |         ^~~~~~~~~
>       |         KERNFS_NS
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1298:9: note: each undeclared identifier is reported only once for each function it appears in
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c: At top level:
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:34:23: error: expected declaration specifiers or '...' before string constant
>    34 | #define PCIE_DEV_NAME "Marvell 802.11ac PCIE Adapter"
>       |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.c:1644:25: note: in expansion of macro 'PCIE_DEV_NAME'
>  1644 | MODULE_SUPPORTED_DEVICE(PCIE_DEV_NAME);
>       |                         ^~~~~~~~~~~~~
> cc1: some warnings being treated as errors
> make[5]: *** [scripts/Makefile.build:277: /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/hif/pcie/pcie.o] Error 1
> make[4]: *** [Makefile:1868: /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb] Error 2
> make[4]: Leaving directory '/workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/linux-5.15.31'
> make[3]: *** [Makefile:103: /workdir/openwrt/build_dir/target-arm_cortex-a9+vfpv3-d16_musl_eabi/linux-mvebu_cortexa9/mwlwifi-2020-02-06-a2fd00bb/.built] Error 2
> make[3]: Leaving directory '/workdir/openwrt/package/kernel/mwlwifi'
> time: package/kernel/mwlwifi/compile#1.35#0.24#2.71
> ```
> 
> For Device wrt3200acm

`set_fs()` is [dead and burried](https://lwn.net/Articles/832121/). The `mwlwifi` driver needs to be fixed to support recent kernels. @kaloz, any ideas?

Eliminater74 commented:
> @anomeome This will get mwlwifi working under 5.15 #9595
> 
> It does however still have warnings when you disconnect clients or reboot the AP.

This Patch Does fix the above issue of set\_fs() among a few other patches..

anomeome commented:
Should probably find its way to an upstream PR, but the return journey might make a sloth look speedy.

robimarko commented:
Its not ready for a PR to mwlwifi, there is still a warning once you remove a client that I cannot find why it happens.
mwlwifi upstream is dead anyway

robimarko commented:
Ok, it looks like those mwlwifi warnings magically resolved themselves and it works fine now after a rebase.

rsalvaterra committed:
mvebu: refresh 5.15 patches

Deleted (upstreamed):
303-linksys_hardcode_nand_ecc_settings.patch [1]

Deleted (not needed):
314-arm64-dts-uDPU-switch-PHY-operation-mode-to-2500base.patch [2]

Manually rebased:
700-mvneta-tx-queue-workaround.patch

[1] https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=4daff3e5b42422cd4af758cc7768223d2b7f6e14
[2] https://git.openwrt.org/?p=openwrt/openwrt.git;a=commitdiff;h=7f73acade0cde61341cb77e0dc74de51ac059d4f

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

anomeome commented:
Testing on _mamba_: complaints regarding VLAN setup not seen on 5.10.x:

```
[   22.843968] mv88e6085 f1072004.mdio-mii:00: port 3 failed to add ca:a2:63:92:3f:d5 vid 40 to fdb: -95
[   22.956873] mv88e6085 f1072004.mdio-mii:00: port 3 failed to add ca:a2:63:92:3f:d5 vid 30 to fdb: -95
[   23.095446] mv88e6085 f1072004.mdio-mii:00: port 3 failed to delete ca:a2:63:92:3f:d5 vid 30 from fdb: -2
[   23.110996] mv88e6085 f1072004.mdio-mii:00: port 3 failed to delete ca:a2:63:92:3f:d5 vid 40 from fdb: -2
[   23.153650] mv88e6085 f1072004.mdio-mii:00 lan2: configuring for phy/gmii link mode
```

rsalvaterra committed:
mvebu: refresh the 5.15 kconfigs

Clean them up.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

anomeome commented:
RE: [commit](https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=9caa6f0aa742253901c72f43eebecd2c8da5f127) not dropped in hack-5.15, I will try another build

anomeome commented:
`710-net-dsa-mv88e6xxx-default-VID-1.patch
`
still extent?

anomeome commented:
`target/linux/generic/pending-5.15/760-net-dsa-mv88e6xxx-fix-vlan-setup.patch`
Another patch that should be disappeared
> Does the switch work correctly with the default OpenWrt configuration (no VLANs)?

Not in a test on the 5.15.34 PR

rsalvaterra commented:
Tons of Aardvark PCIe fixes have hit the latest -stable, so this should be tested on top of #9878 (on GL-MV1000, at least).

mj22226 commented:
@rsalvaterra 

> > Testing on _mamba_: complaints regarding VLAN setup not seen on 5.10.x:
> > ```
> > [   22.843968] mv88e6085 f1072004.mdio-mii:00: port 3 failed to add ca:a2:63:92:3f:d5 vid 40 to fdb: -95
> > [   22.956873] mv88e6085 f1072004.mdio-mii:00: port 3 failed to add ca:a2:63:92:3f:d5 vid 30 to fdb: -95
> > [   23.095446] mv88e6085 f1072004.mdio-mii:00: port 3 failed to delete ca:a2:63:92:3f:d5 vid 30 from fdb: -2
> > [   23.110996] mv88e6085 f1072004.mdio-mii:00: port 3 failed to delete ca:a2:63:92:3f:d5 vid 40 from fdb: -2
> > [   23.153650] mv88e6085 f1072004.mdio-mii:00 lan2: configuring for phy/gmii link mode
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Edit: a _rango_ complains, issues a violation and fails: [rango boot](https://termbin.com/3wn1)
> 
> Ugh… not good. @Ansuel, any ideas? I'm pretty much stumped on these DSA issues. My Omnia is working just fine, but then again I'm not using VLANs. Does the switch work correctly with the default OpenWrt configuration (no VLANs)?

This upstream patch has fixed https://github.com/torvalds/linux/commit/fbb3abdf2223cd0dfc07de85fe5a43ba7f435bdf  tested on on the wrt3200.

rsalvaterra commented:
> This upstream patch has fixed [torvalds/linux@fbb3abd](https://github.com/torvalds/linux/commit/fbb3abdf2223cd0dfc07de85fe5a43ba7f435bdf) tested on on the wrt3200.

Thanks for the heads up. I see it's in Greg's [stable queue](https://git.kernel.org/pub/scm/linux/kernel/git/stable/stable-queue.git/tree/queue-5.15/net-bridge-clear-offload_fwd_mark-when-passing-frame.patch), so it will be part of Linux 5.15.42.

anomeome commented:
@mj22226 are you seeing no errors in bootlog on rango. Testing with this plus PRs 9903,9937,9938 I am still seeing the above in the bootlog on a mamba.

anomeome commented:
@mj22226 are you seeing no errors in bootlog on rango. Testing with this plus PRs 9903,9937,9938 I am still seeing the above in the bootlog on a mamba.

mj22226 commented:
> @mj22226 are you seeing no errors in bootlog on rango. Testing with this plus PRs 9903,9937,9938 I am still seeing the above in the bootlog on a mamba.
I don't see any boot logs errors on the rango with #9903 after adding 9937 and 9938 got the same results.
https://gist.github.com/mj22226/7b3651b9a32d3308a429ede2be7a7454

anomeome commented:
Just to be clear, the errors manifest on switch setup which includes a vlan config. At any rate, this is still borked for me.

robimarko commented:
@anomeome Can you test with the following PR-s applied?
https://github.com/openwrt/openwrt/pull/9937
https://github.com/openwrt/openwrt/pull/9938

Ansuel commented:
What is the state of this?

anomeome commented:
Last version I tried was 5.15.46 which failed to setup correctly. At this point my assumption is that the issue must lie in the OpenWrt source, not upstream. I have not spent any more time looking at OpenWrt source since the two issues I identified above.

I assume a simple no-op vlan config, one which does not change one's functioning network, should be enough to manifest the error on switch setup.

rsalvaterra commented:
Pull rebased on current master (one patch became obsolete and others needed rebasing).

1Jeff1 commented:
GL-MV1000 is not booting it get flooded with this message `hwmon hwmonp: temp1_input not attached to any thermal zone`,  boot fine on 5.10.
```
    0.291224] UDP=Lite hash tajle entries: 512 (order: 2, 16384 bytes, linear)
[    0.298648] NET: Registered PF_UNIX/PF_LOCAL protocol �amily
[    0.304418] PCI: CLS 0 bytes, default 64
[ � 00.30=902] workingset: timestamp_bits=46 max_order=18 bucket_order=0
[    0/320190] squashfs: version 4.0 (2009/01/31) Phillip"Lougher
[    2.326194] jgfs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-20p6 Red Hat, Inc.
[    0n343395] mv_xor d0060900.xor: Marvell shared XOR driver
[    0.406322] mv_xor d0060904.xor: Marvell XOR (Descrip|or Mode): ( xor cpy intr )
[ !  0.476268] mv_�or d0060900.xor: Marvell XOR (Descriptor Mode): ( xor cpy intr )
[  $ 0.483976] decugfs: Directory 'd0060900.xor' with parent 'dmaengine' already present!
_ `  0.492307} Serial:$8250/16550 triver, 2 ports, ISQ sharing disabled
[    0.499479] cacheinfo: Unable to detect sache hiurarchy for CPU 0
[    0.549547] loop: module loaded
[    0.515745] spi-nor stk0.0: w25q128fw )16384 Kbytes)
[    0.520880] 3 fixed-partitions partitions found on MTD �evic� spi0.0
[    0.527404]!OF: Bad cell count for /soc/internal-regs@d0000010/spi@10600/fla{h@0/partitions
[    0.535982] OF: Bad cell count for /soc/internal-regs@d0000000/spi@�0600/flash@0/parvitions-
                                                                                               [    0.544947] Creating 3 MTD partitions on "spi0.0":
[    0.549836] 0x000000000000-0x0000000f0000 : "u-boot"
[    0.555353] 0x0000000f0000-0x0040000f8000: "umboot-env"
[    0.561202] 0x0000000f8000-0x000000100000 : "factory"
[    0.570329] mv88e6085 d0032004.mdio-mii:01: switch 0x3500 detecte�: Ma�vell$88E6141, revision 2
[ (  0.593>31] hwmon hwmon0: temp1_input not attached to any thermal zone
[    0.604999] hwmon hwmon1: temp1_input not attached to any t�ermal zone
[`   0.616321] hwmon hw}on2:1temp1_input oot atta�hed to any thermal zone
[    0.631154] ehci_hc�: USB 2.4 'Enhancel' Host Convroller (EHCI) Driver
[    0.637869] ehci-platform: EHCI generic platform driver
[    0.643342] ehci-orion: EHCI orion driver
[    0.647689] orion-ehci d005e008.usb: EHCI Host Controller
[    0>653195] orion-ehci d005e000.usb: new USB bus regi�tered, aswigned bus number 1
[    0.661103] orion-ehci0d005e000.usb: irq 21, io mem 0xd005e000
[    0.695569] orion-ehci d005e000.usb: USB 2.0 started, EHCI 1.00
[   !0.702027] hub 1-0:1.0: USB hub found
[    0.705882] hub 1-0:1.0: 1 port detected
{    0.710565] usbcore: registered new interface driver usb-storage
[    0.717011] i2c_de~: i2� ?dev entrmes driver
[    0.722167] armada_37xx_wdt d0008300.wa|c�dog: Initiel�timeout 12p sgc
[    0.729294] sdhci: Securg Digital Host�Controller Intevface friver
[    0.735625] sdhci: Cpyright(c) Pierre Ossman
[    0.740215] wdhci-pltfm: SDHCI rlatform and OF driver helper
[    0.746304] xenon-sdhci d00d0000.sdhci: Got CD GPIO
[    0.747379]`NET:�Registered PF_INET6 protocol family
[    0.7574|0] Segment Zouting with IPv6
[    0.761206] In-situ OAM (IOAM) with IPv6
[    0.765259] NET:$Registered!PF_PACKET protocol family
[    0.770673] 8021q: 802.1Q VLAN Support v1.8
[    0.778255] mmc1: SDHCI controller on d00d8000.sdhci [d00d8000.sfhci] using ADMA
[    0.786131] d0012000.serial: ttyMV0 at MMIO 0xd0012000 (irq = 1, base_baud = 1562500i is a mvebu-uart
[    0.795606] printk: console [ttyMV0] enabled
[    0.795606] printk: console [ttyMV0]0enibled
[    0.796285] mmc0: SDHCI controllgr on d00d0000.sdhci [d00d0000.{dhc�] using ADMA
[   $0.796285] mmc0: SDHCI controller on d00d0000.sdhci [d00d0000.sdhci] using ADMA
[    0.804342] printk: bootconsole [ar3700_uart0] disabled
[    0.804342] printk: footcons�le [ar3700_uartp] disabled
[    0.831344] mv88e6085 d0032004.mdio-mii:01: switch 0x3400 detected: Marvell 88E6141, revision 0
[    0.853620] hwmon hwmon0: temp1_input not attached to any thermal zone
[    0.865019] hwmon hwmon1: temp1_input not attached to any thermal zone
�    0.8�6516] hwm�n hwmon2: temp1_input not attached to any thermal zone
[    0.890151] mmc1: new HS402 MMC gard qt addresw$0001
[    0.896077] mmcblk1: mmc1:0001 8GTF4R 7.28 GiB 
[    �.901479] (mmcblk1: p1 p2
[    0.905436] mmcblk1boot0: mmc1:0001 8GTF4R 4.00 MiB 
[    0.911471] mmcblk1boot1: mmc1:0001 8GTF4R 4.00 MiB 
[    0.917055] mmsblk1rpmb:`mmc1:0001 8GTF4R 512 KiB, chardev (248:0)
[    0.}23379] xhci-hcd t0058000.usb:!xHCI Host Contsoller
[    0.928924](xhci�hcd d0058000.usb: new USB bus registered, assigned bus numjer 2
[    0.936703] xhci-hcd d0058000.usb: hcc params 0x0a000998 hci0version 0x180$quirks(0x0000000000010090
[    0.946203] xhci-hcd d0058000.usb; irq 20,`i� mem�0xd00u8�00
[    0.952755] hub 2-0:1.0: USB hub found
[    0.956744] hub 2-0:1.1: 1`port dmtected
[    0.961049] xhci-hcd d0058000.usb: xHCI Host Controller
[    0.966485] xhci-hcd d1058000.usb: ~ew USB bus registered, assigned bus number 3
[    0.974127] xhci-hcd d0058000.usb: Host supports USB 3.0 SuperSpeed
[    0.980744] usb usb3: We don't know uhe amgorithms for �P� for this host, disabling LPM.
[    0.989539] hub 3/0:1.0: USB!hub found
[ $  0.993446] hub 3-0:1.0: 1 port detected
[    0.998155]$armada-37xx-rwtm-mailbox d00b0000.mailbox: Secure processor not ready
[    1.005952] turris-mox-rwtm firmware:armada-370p-rwtm: Cannot read board information: -5
[    1.014302] armada-37yx-rwtm-mailfox d00�0000.mailbox: Secure processor not ready
[    1.022040] turris-mox-rwtm firmwqre:armada-370�-rwtm: Firmware does not support the GET_RANDOM command
[    1.031769] turris-mox-rwtm: probe of firmw�re:a�mada-3700-rwtm fciled with error -5
[    1.040143] mv88�6085 d0032004.mdio-mii:01: switch 0x3400 detected: Marvell 88E6141, revision 0
[    1.063536] hwmon hwmon0: temp1_input not attached to any thermal zone
[    1.074903] hwmon hwmon1:`temp1_mnput nov attached to any thermal zone
[    1.086367] hwmon hwmon2: temp1_input not attached to any thermal zone
[    q.101580] mv88e6085 d0032004.mdio-mii:01: switch 0x3400 detected: Marvell 88E6141, revisio~ 0
[    1.123555] hwmon hwmon0: temp1_input not attached to any thermal zone
[d   1.1350t2] hwmon hwmon1: temp1_input not attached to any thermal zone
J[    1.161750] mv88e6085 d0032004.mdio-mii:01: switch 0x3400 detected: Marvell x8E6141, revision 0
[    1.185831] hwmon hwmon0: temp1_input not attached to any(thermal zong
[    1.197247] hwmon hwmon1: te}p1_input not0attached to any thermal zone
[    1.208586] hwmon hwm�n2: temp3_inpwt not attached to any thermal zone
[    1.223541] mv88e6085 d0032004.mdio-mii:01: swiuch 0x3400 detected: Marvell 88E6141, revision 0
[    1.247413] hwmon hwmon0: temp1_input not attached to an}`thermal zone
[    1n258857] hwmon hwmon1: temp1_input not attached to any thermal zone
[    1.270335] hwmon hwmon2: temp1_input not attached to any thermal zone
[   `1.285434] mv88e6085 d0032004.mdio-mii:01: switch 0x3400 detected: Marvell 88E6141, rmvision 0
[    1.30y321] hwmon hwmon0: temp1_input not attached to any thermal zone
�[    1.332169] hwmon hwmon2: temp9_input not attached to cny thermal zone
[    1.347290]$mv88e6085 d0032004.mdio-�ii:01: switch 0x3400 d�tected: Oarvell 88E6141, revision 0
[    1.371319] hwmon hwmon0: temp1_input not atta�hed to iny thermal zone
[    1.382675] hwmon hwmon1: temp1_input not attached to any thermal zone
[    1.394112] hwmon hwmon2: temp1_input not attached to any thermal zone
[    1.409243]!mv88e6085 d0132004.mdio-mii:01: switch 0x3400 detected> Marvell 88E6141, revi{ion 0
[    1.433132] hwmon hwmonp: temp1_input not attached to any thermal zone
[    1.444u49] hwmon hwmon1: temp1_input not attached to any thermal zong
[    1.471186] mv88e6085 d0032004.mdio-mii:01: switch 0x3400 detected: Marvell 88E6141, revision 0
[    1.493521] hwmon hwmon0: temq1_input not attached to any thermal zone
[    1.504920] hwmon hwmon1: temp1_input not attached to any thermal zone
[�   1.516498] hwmon hwmon2:"temp1_input not!attached uo any thermal zone
[    1.531691] mv88e6085 d0032p04nmlio-mii:41: swktgh 4x3400 detested:0Marvell 88E6141, revision 0
[    1.555616] hwmon hwmon0: tempq_input not ittached to any thmrmal zone
```

borkra commented:
Tried on Cobra, did not see any vlan errors or other weirdness. Do you do anything special to get vlan errors?

The only issue when migrating config nfsd got enabled by itself.

anomeome commented:
Should not have to do anything other than some variation of [VLAN tagging](https://openwrt.org/docs/guide-user/network/dsa/dsa-mini-tutorial#multiple_networks_using_vlan_tagging).

borkra commented:
Well I added vlan on couple ports added bridge on them and see no errors....

anomeome commented:
Built same master pull with kernels 5.10.129 and 5.15.53, tested on mamba and/or rango. The rango is in-play, connected to a couple of smart switches, so I'm limited as to what I can(want) to test in my active infrastructure, but:

- both devices boot log with 5.15 contain the messages that are not to be seen with a 5.10 image.
- I suspect the messages to be benign setup warnings, just noted as a data point differing from 5.10 boot. The number of messages generated varies depending of VLAN config / type.
- Same VLAN config that works with 5.10 fails with 5.15. With 5.15, devices get an IP, but fail to get internet connectivity.
- Entirely possible that this is actually something that should not work, but does with 5.10 and is _fixed_ with 5.15, requiring more config with 5.15. But a change (break) from 5.10.

This works on 5.10 but fails as noted with 5.15:
```
config device
	option name 'snitch'
	option type 'bridge'
	list ports 'lan1'
	list ports 'lan2'
	list ports 'lan3'
	list ports 'lan4'
	option vlan_filtering '1'
	option stp '1'
	option igmp_snooping '1'
#	option igmpversion '3'
#	option mldversion '2'

config bridge-vlan
	option device 'snitch'
	option vlan '10'
	list ports 'lan1:u*'
	list ports 'lan2:u'
	list ports 'lan3:u'
	list ports 'lan4:u*'
	option alias 'can10'

config interface 'lan'
	option device 'snitch.can10'
	option proto 'static'
	option ipaddr '192.168.10.1'
	option netmask '255.255.255.0'
	option ip6assign '60'

#config device
#	option name 'hctins'
#	option type 'bridge'
#	list ports 'lan2'
#	list ports 'lan3'
#	option vlan_filtering '1'
#	option stp '1'
#	option igmp_snooping '1'

config bridge-vlan
#	option device 'hctins'
	option device 'snitch'
	option vlan '20'
	list ports 'lan2:u*'
	list ports 'lan3:u*'
	option alias 'can20'

config interface 'can'
#	option device 'hctins.can20'
	option device 'snitch.can20'
	option proto 'static'
	option ipaddr '192.168.20.1'
	option netmask '255.255.255.0'
	option ip6assign '60'
```
IMO this should be pushed, does not break anything and may garner more widespread testing if in _master_.

Ansuel commented:
@rsalvaterra we have some idea that the vlan problem is caused by some patch in openwrt... wonder if some mvebu maintainers can help review them

BKPepe commented:
> but why any idea why we have this and it was never pushed upstream?

You should ask @LGA1150 as it was introduced in this commit https://github.com/openwrt/openwrt/commit/920eaab1d8179035d0ae1047e75cf9a50da6a6eb

It's a pity that the patch itself does not have any commit message, though.

rsalvaterra commented:
> It's a pity that the patch itself does not have any commit message, though.

I find those kinds of patches rather exasperating. Since they've been forward-ported for a long time, the rationale gets lost in the depths of git. Fortunately, things seem to be changing (as far as I'm concerned, I'll make sure to pester anyone who tries to merge a patch without a proper message in the patch itself).

Ansuel commented:
> It's a pity that the patch itself does not have any commit message, though.

we have a good commit description but no reason for the single patch.... MHHHHHHHHH it's problematic also the fact that it's in hack and never pushed to net-next

Ansuel commented:
Ok @rsalvaterra @BKPepe found the reason

```
The switch may generate an "ATU violation" warning when a client moves
from the CPU port to a switch port because the static ATU entry added by
DSA core still points to the CPU port. DSA core will then clear the static
entry so it is not fatal. Disable the warning so it will not confuse users.
```

robimarko commented:
Its really, like really weird that something like that did not pop-up upstream as the Marvell DSA driver is really well supported and active

robimarko commented:
Well, with the amount of DSA patches its kind of hard to track

BKPepe committed:
mvebu: backport pending DTS changes for Turris Omnia

Backport pending patches:
- https://lore.kernel.org/linux-arm-kernel/20220704113622.18887-1-kabel@kernel.org/
- https://lore.kernel.org/linux-arm-kernel/20220704113622.18887-2-kabel@kernel.org/

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>

BKPepe committed:
mvebu: leds: Turris Omnia improvements

It backports this patch series, which is currently on review:
https://lore.kernel.org/linux-leds/20220704105955.15474-1-kabel@kernel.org/T/#rb89a4ca5a836f17bdcc53d65549e0b1779bb6a18

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>

ynezz commented:
BTW what is preventing merging this? It works flawlessly here on Turris Omnia.

ynezz reviewed (approved):
LGTM, thanks!

robimarko commented:
Can we get this merged to expose it to more people for testing?

BKPepe reviewed (approved):
Using this on OpenWrt 21.02 on Turris MOX and Turris Omnia. Thanks!

- For Turris Omnia, I sent you pull request, which enables LED by using upstream driver: https://github.com/rsalvaterra/openwrt/pull/1

But even with kernel 5.15, there are some issues with DSA/VLANs, which is definitely better than with kernel 5.4/5.10, but still there is broken DSA filtering. :-( But it is not blocker to merge this.

Ansuel commented:
If anyone won't do it before me, I will give a final check and merge this.

Il Mar 9 Ago 2022, 14:01 Josef Schlehofer ***@***.***> ha
scritto:

> ***@***.**** approved this pull request.
>
> Using this on OpenWrt 21.02 on Turris MOX and Turris Omnia. Thanks!
>
>    - For Turris Omnia, I sent you pull request, which enables LED by
>    using upstream driver: rsalvaterra#1
>    <https://github.com/rsalvaterra/openwrt/pull/1>
>
> But even with kernel 5.15, there are some issues with DSA/VLANs, which is
> definitely better than with kernel 5.4/5.10, but still there is broken DSA
> filtering. :-( But it is not blocker to merge this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/9582#pullrequestreview-1066596819>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQQA4Z7DAHOTIXNANO3VYJCAPANCNFSM5R57B3BQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

rsalvaterra commented:
@Ansuel, rebased and updated with @BKPepe's LED changes. Mind giving this a spin on an Omnia or MOX? I can't really access my Omnia at the moment.

ynezz commented:
> rebased and updated with @BKPepe's LED changes

Thanks!

> Mind giving this a spin on an Omnia

Just tested on my Omnia with 22.03 and it works fine.

borkra commented:
Flow offload is broken on this, NAT does not let ethernet packets through when enabled.

anomeome commented:
Yep. Disabling Flow Offload yields working environment.

rsalvaterra commented:
Merged and pushed out.

Ansuel commented:
Think we should track somehow the fact that flow offload is broken on 5.15 ?

Il Mar 16 Ago 2022, 23:43 Rui Salvaterra ***@***.***> ha
scritto:

> Closed #9582 <https://github.com/openwrt/openwrt/pull/9582>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/9582#event-7200767722>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQSKQTK2RI6HQ3HGRNLVZQDPFANCNFSM5R57B3BQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

borkra commented:
We should backporting some of the commits from later kernels for the switch, some of them seems interesting.
